src/main/helm/exec.ts

```diff
@@ -7,14 +7,23 @@ import { promiseExecFile } from "../../common/utils/promise-exec";
 import type { BaseEncodingOptions } from "fs";
 import type { ExecFileOptions } from "child_process";
 import { helmBinaryPath } from "../../common/vars";
+import { UserStore } from "../../common/user-store";
 
 /**
  * ExecFile the bundled helm CLI
  * @returns STDOUT
  */
 export async function execHelm(args: string[], options?: BaseEncodingOptions & ExecFileOptions): Promise<string> {
   try {
-    const { stdout } = await promiseExecFile(helmBinaryPath.get(), args, options);
+    const opts = { ...options };
+
+    opts.env ??= process.env;
+
+    if (!opts.env.HTTPS_PROXY && UserStore.getInstance().httpsProxy) {
```

Comment:
Agreed, though I think that be done when this code is made DI complaint.

Reply:
Not sure I understand. You mean if someone copies this code to add proxy support elsewhere (possibly unrelated to helm)? It is flexibility in case the caller of `execHelm` actually wants different behaviour. Maybe not necessary.